### DISCUSSION TITLE
Create Dynamic Jobs from class name as string

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1638] [Q&A] [answered]
Hello, Still new to quartz and I'm having trouble figuring out how to create jobs from a string.

Here is a sample example that doesnt't work.

         
            var _f = Type.GetType("Utilities.TestJob,Utilities")

            IScheduler scheduler = (IScheduler)factory.GetScheduler();

            //...check if a specific job exists
            IJobDetail job = JobBuilder.Create<_f>()  // issue with _f
                .WithIdentity("myJob", "group1")
                .UsingJobData("jobSays", "Bingo")
                .UsingJobData("myFloatValue", 3.141f)
                .StoreDurably(true).Build();

            job.JobDataMap.Add("test", 123);

            scheduler.AddJob(job, true);
            //if yes then run that job
            scheduler.TriggerJob(new JobKey("myJob", "group1"));

As you can see I need to be able to add jobs on the fly and don't want to hard code in the Job Class.

**Comment by lahma** [comment, marked as answer]
`JobBuilder.Create` has an overload to take a type, so it should be enough to use `JobBuilder.Create(_f)`.

**Reply by CrazyMoka** [reply]
Problem is there is an error doing JobBuilder.Create(_f)  it states that _f is a variable but is used like a type.

**Reply by lahma** [reply]
Please check your syntax,  it should work.

**Reply by CrazyMoka** [reply]
Thaks it worked.
